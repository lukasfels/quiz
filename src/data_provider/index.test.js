import { expect, test } from 'vitest'
import { dataProvider, assetUrl, filterProvider } from './'

test('assetUrl: return null if mode does not exist', () => {
  expect(assetUrl('does not exists', null)).toBe(null)
})

test('assetUrl: return string if mode exist', () => {
  stubInputs().forEach(input => {
    expect(assetUrl(input.mode, input.item)).toBeTypeOf('string')
  })
})

test('filterProvider: return hash of categories for specific mode', () => {
  expect(filterProvider('pokemon')).toMatchSnapshot()
})

test('filterProvider: return empty array for no-existent mode', () => {
  expect(filterProvider('nothing')).toMatchObject([])
  expect(filterProvider()).toMatchObject([])
})

function stubInputs() {
  return [
    {
      mode: 'flags',
      item: {
        code: "de",
      }
    },
    {
      mode: 'outlines',
      item: {
        continent: "Afrika",
        code: "zm",
      }
    },
    {
      mode: 'pokemon',
      item: {
        generation: 1,
        id: 25,
      }
    },
    {
      mode: 'pokemon_type',
      item: 'grass'
    }
  ]
}
