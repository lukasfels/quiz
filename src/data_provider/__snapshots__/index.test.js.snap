// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`filterProvider: return hash of categories for specific mode 1`] = `
[
  {
    "categories": [
      {
        "category": "1",
        "collapsed": true,
        "count": 151,
        "index": 0,
        "text": "1. Generation",
      },
      {
        "category": "2",
        "collapsed": true,
        "count": 100,
        "index": 1,
        "text": "2. Generation",
      },
      {
        "category": "3",
        "collapsed": true,
        "count": 135,
        "index": 2,
        "text": "3. Generation",
      },
      {
        "category": "4",
        "collapsed": true,
        "count": 107,
        "index": 3,
        "text": "4. Generation",
      },
      {
        "category": "5",
        "collapsed": true,
        "count": 156,
        "index": 4,
        "text": "5. Generation",
      },
      {
        "category": "6",
        "collapsed": true,
        "count": 72,
        "index": 5,
        "text": "6. Generation",
      },
      {
        "category": "7",
        "collapsed": true,
        "count": 88,
        "index": 6,
        "text": "7. Generation",
      },
    ],
    "title": "Pok√©mon-Generationen",
  },
]
`;
